cmd/login/wc.go

```diff
@@ -147,59 +204,84 @@ func (r *runner) createClusterClientCert(ctx context.Context, client k8sclient.I
 		certOperatorVersion: certOperatorVersion,
 	}
 
-	clientCertResource, err := createCert(ctx, clientCertService, certConfig)
+	clientCertResource, secret, err := r.getCredentials(ctx, services.clientCertService, certConfig)
 	if err != nil {
-		return microerror.Mask(err)
+		return "", false, microerror.Mask(err)
 	}
 
-	// Retrieve client certificate credential.
-	secret, err := fetchCredential(ctx, provider, clientCertService, clientCertResource)
+	credentialConfig := credentialConfig{
+		clusterID:     r.flag.WCName,
+		certCRT:       secret.Data[credentialKeyCertCRT],
+		certKey:       secret.Data[credentialKeyCertKey],
+		certCA:        secret.Data[credentialKeyCertCA],
+		clusterServer: getServerAddress(certConfig),
+		filePath:      r.flag.SelfContained,
+		loginOptions:  r.loginOptions,
+	}
+
+	contextName, contextExists, err := r.storeWCCredentials(credentialConfig)
 	if err != nil {
-		return microerror.Mask(err)
+		return "", false, microerror.Mask(err)
 	}
 
-	// Store client certificate credential either into the current kubeconfig or a self-contained file if a path is given.
-	var contextExists bool
-	var contextName string
-	if r.loginOptions.selfContainedWC {
-		contextName, contextExists, err = printWCCredentials(r.k8sConfigAccess, r.fs, r.flag.SelfContained, clientCertResource, secret, clusterBasePath, r.loginOptions)
+	if clientCertResource != nil {
+		// Cleaning up leftover resources.
+		err = cleanUpClientCertResources(ctx, services.clientCertService, clientCertResource)
 		if err != nil {
-			return microerror.Mask(err)
+			return "", false, microerror.Mask(err)
 		}
-	} else {
-		contextName, contextExists, err = storeWCCredentials(r.k8sConfigAccess, r.fs, clientCertResource, secret, clusterBasePath, r.loginOptions)
+	}
+
+	fmt.Fprint(r.stdout, color.GreenString("\nCreated client certificate for workload cluster '%s'.\n", r.flag.WCName))
+	return contextName, contextExists, nil
+}
+
+func (r *runner) getCredentials(ctx context.Context, clientCertService clientcert.Interface, config clientCertConfig) (*clientcert.ClientCert, *v1.Secret, error) {
+	var secret *v1.Secret
+	var clientCert *clientcert.ClientCert
+	var err error
+
+	// If cert-operator is not running (as in CAPI) we attempt to use the MC PKI to create a certificate
+	if config.certOperatorVersion == "" {
```

Comment:
@tfussell I looked into implementing a `PureCAPI` check here but it would add a lot more complexity to this PR. 
For now, trying to use the MCPKI method in case `cert-operator` is not present makes more sense IMO since that is the reason the old implementation does not work for CAPI releases. WDYT?

Reply:
I see what you mean. Checking for a release label and cert-operator in that release is equivalent to checking if a cluster is "pure CAPI" so I think it's fine.